http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/AppUsageComponent.scala

```diff
@@ -0,0 +1,107 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import cats.effect.Sync
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.leonardo.AppId
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.api._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.dummyDate
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.mappedColumnImplicits._
+import org.broadinstitute.dsde.workbench.leonardo.http.dbioToIO
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.typelevel.log4cats.Logger
+
+import java.sql.SQLDataException
+import java.time.Instant
+import scala.concurrent.ExecutionContext
+
+class AppUasageTable(tag: Tag) extends Table[AppUsageRecord](tag, "APP_USAGE") {
+  def id = column[AppUsageId]("id", O.PrimaryKey, O.AutoInc)
+  def appId = column[AppId]("appId")
+
+  def startTime = column[Instant]("startTime", O.SqlType("TIMESTAMP(6)"))
+  def stopTime = column[Instant]("stopTime", O.SqlType("TIMESTAMP(6)"))
+
+  def * = (id, appId, startTime, stopTime) <> (AppUsageRecord.tupled, AppUsageRecord.unapply)
+}
+
+object appUsageQuery extends TableQuery(new AppUasageTable(_)) {
```

Comment:
Just to make sure I understand, every new start/stop pair will be recorded in a new row in the db correct? Nothing is overwritten?

Reply:
yes..